docs/sdg/knowledge-doc-ingestion.md

```diff
@@ -0,0 +1,96 @@
+# Knowledge Document Ingestion Pipeline Design Proposal
+
+**Author**: Aakanksha Duggal
+
+## 1. Introduction
+
+As part of extending InstructLab's capabilities, this pipeline is designed to support ingestion and processing of various document formats such as Markdown, PDF, DOCX, and more. The goal is to create a unified, modular system that seamlessly integrates with **Synthetic Data Generation (SDG)** and **Retrieval-Augmented Generation (RAG)** workflows, simplifying the process for users while maintaining flexibility for future enhancements.
+
+## 2. Use Case
+
+To enable the ingestion and processing of a wide range of document types, this pipeline must handle formats including:
+
+- Markdown (MD)
+- PDF
+- TXT
+- ASCII Docs
+- DOCX
+- HTML
+- PPTX
+
+This proposal outlines how to build a pluggable system that accommodates these formats and integrates effectively into existing **SDG** and **RAG** workflows.
+
+## 3. Proposed Approach
+
+### 3.1 Custom InstructLab Schema Design
+
+- **Objective**: Define a custom "instructlab" schema that standardizes input formats for SDG and RAG pipelines. This schema will bridge the gap between various document types and the specific formats required for further processing.
+- **Modularity**: The system should allow easy extension, enabling support for new document types and processing workflows without disrupting core functionality.
+  
+The instructlab schema will serve as the intermediary format, supporting flexibility while ensuring compatibility with the ingestion process.
+
+### 3.2 PDF and Document Conversion via Docling
+
+- **Docling Integration**: We will leverage **Docling** to convert files into structured JSON, which will be the starting point for the instructlab schema. Individual components will post-process this JSON as per the requirements of the specific SDG and RAG workflows.
```

Comment:
Could we include a link here to somewhere where folks can read more about Docling, and perhaps a bit as to why Docling is the chosen solution here?